### PR TITLE
Fix "reveal civilians" not resetting civilians to neutral

### PR DESCRIPTION
**Pull request by Hdt80bro**
Got broken in the recall patch; this fixes it

### CODE REVIEW

**Comment by Hdt80bro** [comment]
Why don't AI's get to see civilians?

**Comment by Garanas** [comment]
I don't know, @relent0r would you want AIs to be able to see civs like humans can?

**Comment by relent0r** [comment]
I actually got sick of AI being unable to see civilians a while back and did a little hack that would reveal civilians (set them as allies) for half a second then revert.

So yes I'd love for AI to be able to see Civilians by default as currently without the above they just suicide a bunch of early units into PDs.

**Comment by Hdt80bro** [comment]
Oh, well it'd be trivially easy to get them to see civilians--we actually explicitly exclude them from the reveal.

**Comment by relent0r** [comment]
Ha that sounds like discrimination :) and an odd decision to make. I'm not sure if there would be any consequences for coop games though, that's the only reason I would imagine they would be excluded.

**Comment by Garanas** [comment]
> Ha that sounds like discrimination :) and an odd decision to make. I'm not sure if there would be any consequences for coop games though, that's the only reason I would imagine they would be excluded.

That is an interesting question, we'll find out I assume.

File: lua/sim/ScenarioUtilities.lua [hunk 1]
```diff
@@ -618,11 +617,11 @@ function InitializeArmies()
 
             local armyIsCiv = setup.Civilian
 
-            if armyIsCiv and civOpt ~= 'neutral' and strArmy ~= 'NEUTRAL_CIVILIAN' then -- give enemy civilians darker color
+            if armyIsCiv and civOpt ~= "neutral" and strArmy ~= "NEUTRAL_CIVILIAN" then -- give enemy civilians darker color
```
Comment (Garanas) [inline lua/sim/ScenarioUtilities.lua:620]: Is the lua plugin telling you to make these changes?

File: lua/sim/ScenarioUtilities.lua [hunk 2]
```diff
@@ -728,7 +722,7 @@ function InitializeScenarioArmies()
             local setup = armySetups[strArmy]
             local brain = GetArmyBrain(strArmy)
 
-            LOG('*DEBUG: InitializeScenarioArmies, army = ', strArmy)
+            LOG("*DEBUG: InitializeScenarioArmies, army = ", strArmy)
```
Comment (Garanas) [inline lua/sim/ScenarioUtilities.lua:725]: This LOG statement can go

File: lua/sim/ScenarioUtilities.lua [hunk 3]
```diff
@@ -606,7 +605,7 @@ function InitializeArmies()
             local setup = armySetups[strArmy]
             local brain = GetArmyBrain(strArmy)
 
-            --LOG('*DEBUG: InitializeArmies, army = ', strArmy)
+            --LOG("*DEBUG: InitializeArmies, army = ", strArmy)
```
Comment (Garanas) [inline lua/sim/ScenarioUtilities.lua:608]: This comment can go too

**Review by Garanas — APPROVED** [review]

File: lua/sim/ScenarioUtilities.lua (hunk 1 above)
Comment (Hdt80bro) [reply lua/sim/ScenarioUtilities.lua:620]: No, that's just my personal preference =)

**Pull request merged**